dali/operators/generic/slice/slice_attr.cc

```diff
@@ -19,6 +19,31 @@
 
 namespace dali {
 
+void ProcessAxesArgs(std::vector<int> &axes, TensorLayout &axis_names, const OpSpec &spec,
+                     const char* axes_arg_name, const char* axis_names_arg_name) {
+  axes.clear();
+  axis_names = TensorLayout{};
+  const bool use_axes = axes_arg_name != nullptr;
+  const bool use_axis_names = axis_names_arg_name != nullptr;
+  const bool has_axes_arg = use_axes && spec.HasArgument(axes_arg_name);
+  const bool has_axis_names_arg = use_axis_names && spec.HasArgument(axis_names_arg_name);
+  if (has_axes_arg && has_axis_names_arg) {
+    DALI_FAIL(make_string("\"", axes_arg_name, "\" and \"", axis_names_arg_name,
+                          "\" are mutually exclusive"));
+  } else if (has_axes_arg) {
+    axes = spec.GetRepeatedArgument<int>(axes_arg_name);
+    axis_names = TensorLayout{};
+  } else if (use_axis_names && spec.TryGetArgument(axis_names, axis_names_arg_name)) {
+    axes = {};
+  } else if (use_axes && spec.TryGetArgument(axes, axes_arg_name)) {
+    axis_names = TensorLayout{};
+  } else {
+    // Will use all-axes
+    axes = {};
```

Comment:
There should be a dedicated flag for that so that we can distinguish:
`axes=None` or `axis_names=None` (use all)
from 
`axes=[]` or `axis_names=""` (no axes - no slicing)

Reply:
done